src/main/java/bio/terra/filesystem/FireStoreDependencyDao.java

```diff
@@ -34,9 +34,19 @@ public FireStoreDependencyDao(Firestore firestore, FireStoreUtils fireStoreUtils
         this.fireStoreUtils = fireStoreUtils;
     }
 
-    public boolean hasDatasetReference(String studyId, String objectId) {
+    public boolean objectHasDatasetReference(String studyId, String objectId) {
         CollectionReference depColl = firestore.collection(DEPENDENCY_COLLECTION_NAME);
         Query query = depColl.whereEqualTo("studyId", studyId).whereEqualTo("objectId", objectId);
+        return hasReference(query);
+    }
```

Comment:
I think we should return a list of the datasets that reference the object. same for study below

Reply:
I would say when we need that interface, we can add it. None of the current use needs it.

Reply:
i think that is usually the information that the user wants. If they are trying to delete something but a dependency is blocking them, they are going to want to know what is preventing it. it would be useful to return that data in the exception.

Reply:
Sure, I understand. I don't think pushing that info through the exception is the right answer.
I think we should have a way to view dependencies separate from the delete logic.

Reply:
i agree that we should have a way to check dependencies that isn't tied to deleting. but i think that when a delete fails because of dependencies our users will want that information.

Reply:
I agree that people may want the info. I disagree that returning an unbounded query as free text in an error message is the right way to solve the problem.

If we do this we will have to remember we are returning query results that may need to be authorized as the permission model evolves.